lua/neo-tree/defaults.lua

```diff
@@ -22,6 +22,7 @@ local config = {
   enable_modified_markers = true, -- Show markers for files with unsaved changes.
   enable_opened_markers = true,   -- Enable tracking of opened files. Required for `components.name.highlight_opened_files`
   enable_refresh_on_write = true, -- Refresh the tree when a file is written. Only used if `use_libuv_file_watcher` is false.
+  enable_normal_mode_for_inputs = true, -- Enable normal mode for input dialogs.
```

Comment:
I think the default should be `false` for this option.

We should also include this option in the quickstart of the README.

Reply:
I've made the requested changes, but I have another concern about the input box for the `delete file` operation. I believe it's not necessary to revert to normal mode since it only requires a `y` or `n` input. 

Here's my suggested code update for the confirm input to always retain the old behavior:

```lua
  if config.enable_normal_mode_for_inputs and input.prompt_type ~= "confirm" then
    vim.schedule(function()
      vim.cmd("stopinsert")
    end)
  end
  
  input:map("i", "<esc>", function()
    vim.cmd("stopinsert")
    if not config.enable_normal_mode_for_inputs or input.prompt_type == "confirm" then
      input:unmount()
    end
  end, { noremap = true })

  ```
  
  By adding `input.prompt_type = "confirm"` to the M.confirm function below before calling `M.show_input(input)`, we can handle this. Alternatively, we can check `input._.prompt._content == " y/n: "` without adding a value to the input.

If you think the current implementation is fine, I'm okay with that as well.

Reply:
I agree that y/n prompts should be in insert mode regardless of the `enable_normal_mode_for_inputs` setting. I also think adding `input.prompt_type = "confirm"` is the best way to accomplish that.